autoprecompiles/src/optimizer.rs

```diff
@@ -112,60 +105,79 @@ fn solver_based_optimization<T: FieldElement>(
     result.simplified_constraint_system
 }
 
-fn remove_trivial_bus_interactions<T: FieldElement>(
+/// Removes any columns that are not connected to *stateful* but interactions (e.g. memory),
+/// because those are the only way to interact with the rest of the system.
+/// We assume that all input constraints are satisfiable. Because they are not connected to
```

Comment:
What about 
```
x = y + 1
y = x + 1
```
Each constraint is satisfiable, but together they are not?
Maybe a better phrasing would be "any subset of the constraints is satisfiable"? Is is correct?

Reply:
I think it's an assumption that the entire system is not unsat (which would be a completeness bug).

Reply:
sgtm but will defer to @leonardoalt